src/components/ProductCard/ProductCard.jsx

```diff
@@ -1,30 +1,141 @@
+          authorization: token,
+        }
+      },)
+     if(response.status===200)
+       setWished(false);
+      wishDispatch({type:"ADD_TO_WISHLIST", payload:response.data.wishlist})
+    }
+    catch(error){
+      console.log(error);
+    }
+  }
+
+  const addToCartHandler=async(id)=>{
+    if(!token)
+     {
+       navigate("/login");
+     }
+     else{
+       const item= initialProduct.find(element=>element._id===id);
+       
+      try{
+              const response= await axios.post('/api/user/cart',{product:item},
+              {
+                headers:{
+                  authorization: token,
+              },
+            }
+              )
+             
+              if(response.status===201)
+              setCarted(true);
+              dispatch({type:"ADD_TO_CART", payload:response.data.cart})
+           }
+           catch (error){
+             console.log(error);
+           }
+    
+     }
+}
+
+const goToCart=()=>{
+   navigate("/cart");
+}
+  useEffect(()=>{
+    if(wishState.wishListData.some((ele)=>ele._id===item._id))
+      setWished(true);
+    if(state.cartData.some((ele)=>ele._id===item._id))
+    setCarted(true);
+       
+  },[wishState,state])
+
 return (
     
           <div class="card relative">
             <div>
               <section className="relative">
-                  <div className="absolute wishlist-icon "><i className="far fa-heart icon-sm"></i></div>
-                <img src={image} alt="sneaker" className="card-img"/>
+                {!wished?<div className="absolute wishlist-icon "
+                  onClick={()=>addToWishListHandler(item._id)}
+                  ><i className="far fa-heart icon-sm"></i></div>:
+                  
+                  <div className="absolute wishlist-icon "
+                  onClick={()=>removeFromWishListHandler(item._id)}
+                  ><i class="fas fa-heart added-wishlist-icon"></i></div>
+                  }
+
+                  
+                <img src={item.image} alt="sneaker" className="card-img"/>
               </section>
               <section class="sec-2">
                 <div className="flex-row title-rating-container">
-                <h3 className="product-title">{title}</h3>
-                <p className="product-rating">{rating} <i className="fas fa-star"></i></p>
+                <h3 className="product-title">{item.title}</h3>
+                <p className="product-rating">{item.rating} <i className="fas fa-star"></i></p>
                 </div>
                 
-                <p className="price-tag"><strong>Rs. {price}</strong></p>
+                <p className="price-tag"><strong>Rs. {item.price}</strong></p>
               </section>
             </div>
 
-            <section className="sec-3">{description}</section>
+            <section className="sec-3">{item.description}</section>
             <footer>
+              {!carted?<button className="btn productCard-button add-cart-btn flex-row absolute"
```

Comment:
nice logic, but here you can "trim the fat" rather than using ternary operator on the whole element you can use it on the pieces which are changing